### PR TITLE
Improve search of cuda virtual package

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
---

`nvidia-smi` binary is currently used to get the `cuda` driver version.
On Windows, this utility may be not located in the `path`, then:
- check if `nvcuda.dll` is found in the path
- if true, search in all `C:\Windows\System32\DriverStore\FileRepository` subdirs if one starting with `nv` contains `nvidia-smi.exe` binary file

Also lower log message level to `debug` if `cuda` is not found, since it's will be a pretty common case.

### CODE REVIEW

File: src/virtual_packages.cpp [hunk 1]
```diff
@@ -92,7 +92,50 @@ namespace mamba
...
+                    {
+                        may_exist = true;
+                        break;
+                    }
+                }
+
+                std::string base = "C:\\Windows\\System32\\DriverStore\\FileRepository";
+                if (may_exist)
+                {
+                    for (fs::path p : fs::directory_iterator(base))
```
Comment (wolfv) [inline src/virtual_packages.cpp:113]: I think `auto&` is better

File: src/virtual_packages.cpp [hunk 2]
```diff
@@ -92,7 +92,50 @@ namespace mamba
...
+                    {
+                        if (p.string().startswith("nv")
```
Comment (wolfv) [inline src/virtual_packages.cpp:115]: then maybe have to use `p.path().string()...`

File: src/virtual_packages.cpp [hunk 3]
```diff
@@ -92,7 +92,50 @@ namespace mamba
...
+                            && fs::exists(concat(p, env::pathsep(), "nvidia-smi.exe")))
```
Comment (wolfv) [inline src/virtual_packages.cpp:116]: maybe not use concat, but rather `p / "nvidia-smi.exe"`

File: src/virtual_packages.cpp [hunk 4]
```diff
@@ -92,7 +92,50 @@ namespace mamba
...
+                        {
+                            std::string err;
+                            std::string f = fs::exists(concat(p, env::pathsep(), "nvidia-smi.exe"));
```
Comment (wolfv) [inline src/virtual_packages.cpp:119]: ? looks liek a bug (assigning bool to string)

Comment (adriendelsalle) [reply src/virtual_packages.cpp:119]: it is !

**Pull request merged**